src/sanitizer.js

```diff
@@ -8,18 +8,31 @@
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocFragment = function _sanitizeDocFragment(config, input) {
     // TODO implement this function and remove eslint-disable comment above
+    // TODO maybe we can use https://developer.mozilla.org/en-US/docs/Web/API/NodeIterator to traverse through nodes
 };
 
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocument = function _sanitizeDocument(config, input) {
     // TODO implement this function and remove eslint-disable comment above
 };
 
-// a parser that doesn't run scripts or loads resources
-// eslint-disable-next-line no-unused-vars
-export const _fragmentParser = function _fragmentParser(ctx, input) {
-    // note: ctx is of type Node (element?)
-    // TODO implement this function and remove eslint-disable comment above
+/**
+ * fragmentParser
+ * a parser that doesn't run scripts or loads resources
```

Comment:
ouch. I just tested this. That's not actually true :rofl: :sob: 
This is bad, but it doesn't justify throwing all of your other useful code changes away. Can you change that comment and file a follow-up? I have _some_ ideas, that I'll need to verify and will get into that issue once it is filed.

Reply:
> ouch. I just tested this. That's not actually true 🤣 😭

Oh noo, that's right... `Range.createContextualFragment` will execute scripts 😢

I guess we are 11 years too late to take advantage of [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=599588) 🙃

> This is bad, but it doesn't justify throwing all of your other useful code changes away. Can you change that comment and file a follow-up?

Sure! I'll file an issue, update the comment to remove "doesn't run scripts" and add a reference to the issue in code. 

> I have some ideas, that I'll need to verify and will get into that issue once it is filed.

I found an interesting old [WebKit](https://bugs.webkit.org/show_bug.cgi?id=12234#c8 ), [WhatWG](https://lists.whatwg.org/pipermail/whatwg-whatwg.org/2010-November/071276.html) and [Twitter](https://mobile.twitter.com/wesbos/status/1110942279160786944?lang=en) conversations on this. 

Also this post might be useful here - https://grrr.tech/posts/create-dom-node-from-html-string/

Reply:
After some research, I think we just have to use the same techniques DOMPurify is using: Use a new document, disregard the context :-(
An alternative would be to create a mock-context, i.e., adding the context Element to the mock-document and then parsing in there. That won't help when the stack of open elements contains more than 1 required element (would work for `tr` in `table` but maybe not for `td` in `tr`, because we're lacking the outer `table`). Not sure if that's worth for a polyfill, when DOMPurify has so far been almost untouched by this flaw?

Reply:
> After some research, I think we just have to use the same techniques DOMPurify is using: Use a new document, disregard the context :-(

> Not sure if that's worth for a polyfill, when DOMPurify has so far been almost untouched by this flaw?

Ah, should we abandon this PR and go back to shimming DOMPurify?

Reply:
Yeah, I'm afraid we have to...Sorry for this all...

Reply:
No problem at all! I learned a lot from going down this path 😄 

I'll go back to your original plan and start working on it!